include/swift/AST/Evaluator.h

```diff
@@ -349,8 +347,14 @@ class Evaluator {
                          bool lastChild) const;
 
   /// Print the dependencies of the given request as a tree.
-  void printDependencies(const AnyRequest &request,
-                         llvm::raw_ostream &out) const;
+  template <typename Request>
+  void printDependencies(const Request &request, llvm::raw_ostream &out) const {
```

Comment:
Why bother pushing this up into the header? It's not in the critical path at all.

Reply:
I wanted to leave the AnyRequest constructor `explicit`, but everywhere that `printDependencies` is used today it takes a specific request. Other options include:

- write this as a wrapper around the original two-argument `printDependencies`
- just make all callers construct the AnyRequest explicitly

But I thought this one seemed simplest.